gpytorch/distributions/multivariate_normal.py

```diff
@@ -37,7 +36,8 @@ def __init__(self, mean, covariance_matrix, validate_args=False):
             self._covar = covariance_matrix
             self.__unbroadcasted_scale_tril = None
             self._validate_args = validate_args
-            batch_shape, event_shape = self.loc.shape[:-1], self.loc.shape[-1:]
+            batch_shape = _mul_broadcast_shape(self.loc.shape[:-1], covariance_matrix.shape[:-2])
```

Comment:
@gpleiss this broke our multitask models in botorch. Before I dig deeper into this, what was the rationale to add this here?

Reply:
more precisely, the failure is on batch evaluating a non-batched model

Reply:
@Balandat Do you have an example test that fails?

Reply:
This is the behavior that `torch.distributions.MultivariateNormal` has - if you pass in a non-batch mean and a batch covariance matrix, then the covariance matrix determines the batch size. This wasn't happening before.

I believe the use case was for training (independent) multi-output variational models. Let's say you want a different lengthscale for each output, but the same mean. This way, you can train a model with non-batch `x` inputs, a non-batch mean function, and a batch kernel function (with different lengthscale). With this fix, the resulting MVN has the expected batch size.

Reply:
Hmm nvm this was actually a bug in our MTGP implementation that would do the wrong thing in batch model. All good.

Reply:
thanks for surfacing the bug :)